fvm/evm/emulator/emulator.go

```diff
@@ -126,6 +126,7 @@ func (bl *BlockView) DirectCall(call *types.DirectCall) (res *types.Result, err
 				receipt := &gethTypes.Receipt{}
 				if res != nil {
 					receipt = res.Receipt()
+					err = res.ValidationError
```

Comment:
hmm, wouldn't this override the error we are going to return, can we use a different name for it like vmErr

Reply:
we can to be safe, but in case `res != nil` there should be no error, however that's a bit implicit